### ISSUE TITLE
SA1501 should allow delegate assignments, and code fix needs work

### ISSUE THREAD

**Issue by AArnott** [DotNetAnalyzers/StyleCopAnalyzers#855]
SA1501 doesn't like to see this line:

        var foo = () => { return 5; };

And its code-fix turns it into this:

        var foo = () => {
            return 5;
    }
    ;

Note it mis-indents the 3rd line, and puts the semicolon on a line by itself. Also the "fixed" code itself generates an SA1500 diagnostic because the opening curly shares a line with the code before it.

**Comment by AArnott** [comment]
It doesn't like it even if the delegate block is empty.

**Comment by vweijsters** [comment]
Assuming that you mean the following code (your example triggers CS0815 for me):

```C#
Func<int> foo = () => { return 5; };
```

There is indeed a bug in the analyzer. The analyzer should allow lambda expressions and anonymous methods on a single line, as long as the entire expression that it is part of is on the same line, as that is the behavior of the original StyleCop.

I'll have a look at the code-fix as well, it is obviously bugged, but I haven't been able to analyze it yet.

@sharwell assign it to me please.

**Comment by sharwell** [comment]
Note that if the warning is suppressed for this case, the code fix behavior becomes irrelevant. :smile:

**Comment by vweijsters** [comment]
That was my initial thought as well :smile: 

But then I thought of this test case:
```C#
Func<int> foo = () => 
    { return 5; };
```

which will trigger SA1501 and must therefore offer a working code fix, but the codefix now produces the mess that @AArnott described above.

**Comment by AArnott** [comment]
Thanks, @vweijsters
